engine-tests/src/test/java/org/terasology/engine/integrationenvironment/MTEExtensionTestWithPerMethodLifecycle.java

```diff
@@ -68,29 +64,26 @@ public void firstTestFindsThings() {
         assertEquals(1, entities.size());
 
         DummyComponent component = entities.get(0).getComponent(DummyComponent.class);
-        assertTrue(component.eventReceived);
+        assertThat(component.eventReceived).isTrue();
+        assertThat(seenNames.isEmpty());
 
         // Remember that a test has seen this one.
-        assertNotNull(component.name);
-        assertFalse(seenNames.containsKey(component.name));
-        seenNames.put(component.name, 1);
+        assertThat(component.name).isNotNull();
+        assertThat(seenNames).isEmpty();
+        seenNames.add(component.name);
     }
 
     @Test
     @Order(2)
-    public void thingsStillExistForSecondTest() {
+    public void thingsDoNotPolluteSecondTest() {
         List<EntityRef> entities = Lists.newArrayList(entityManager.getEntitiesWith(DummyComponent.class));
         // There should be one entity, created by the @BeforeAll method
```

Comment:
There is no longer a `@BeforeAll` method, but only the `@BeforeEach`. 

```suggestion
        // There should be one entity, created by the @BeforeEach method for this one test
```

Also needs to be updated in 

https://github.com/MovingBlocks/Terasology/blob/1b230c2354a540a54b485d3a333d7627babb8d0a/engine-tests/src/test/java/org/terasology/engine/integrationenvironment/MTEExtensionTestWithPerMethodLifecycle.java#L46

and 

https://github.com/MovingBlocks/Terasology/blob/1b230c2354a540a54b485d3a333d7627babb8d0a/engine-tests/src/test/java/org/terasology/engine/integrationenvironment/MTEExtensionTestWithPerMethodLifecycle.java#L63

Reply:
There's some disorganization among the tests. Not always clear where to find (or add) a test case with the various combinations of features, and some things seem redundant.

The problem with this class is that I re-labelled it as a "does not share data between tests" thing, but the `seenNames` field it's checking in the second part _is_ shared because it's _static,_ so it's not making a lot of sense and it's not real clear what it should do.

One option would be to drop this entire test class and claim it is covered by https://github.com/MovingBlocks/Terasology/blob/1b230c2354a540a54b485d3a333d7627babb8d0a/engine-tests/src/test/java/org/terasology/engine/integrationenvironment/LifecyclePerMethodInjectionTest.java

I think the difference between this test and that one is that this messes with the state of the EntityManager, and the `InjectionTest` one only meddles with the Context.

Is that close enough, or is Context too much of a weird special case compared to the other managers?